File: app/src/main/java/com/vtrack/data/model/FuelEntry.kt
package com.vtrack.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "fuel_entries",
    foreignKeys = [
        ForeignKey(
            entity = Vehicle::class,
            parentColumns = ["id"],
            childColumns = ["vehicleId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [
        Index("vehicleId"),
        Index("vehicleId", "odometer"),
        Index("vehicleId", "date")
    ]
)
data class FuelEntry(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val vehicleId: Long,
    val date: Long,
    val odometer: Int,
    val gallons: Double,
    val pricePerGallon: Double,
    val totalCost: Double,
    val isPartialFill: Boolean = false,
    val notes: String? = null,
    val createdAt: Long = System.currentTimeMillis()
)

File: vtrack-ios/VTrack/Models/FuelEntry.swift
import Foundation
import SwiftData

@Model
final class FuelEntry {
    var vehicle: Vehicle?
    var date: Date
    var odometer: Int
    var gallons: Double
    var pricePerGallon: Double
    var totalCost: Double
    var isPartialFill: Bool
    var notes: String?
    var createdAt: Date

    init(vehicle: Vehicle, date: Date, odometer: Int, gallons: Double, pricePerGallon: Double, totalCost: Double, isPartialFill: Bool = false, notes: String? = nil) {
        self.vehicle = vehicle
        self.date = date
        self.odometer = odometer
        self.gallons = gallons
        self.pricePerGallon = pricePerGallon
        self.totalCost = totalCost
        self.isPartialFill = isPartialFill
        self.notes = notes
        self.createdAt = Date()
    }
}
